pilgen/src/lib.rs

```diff
@@ -49,6 +51,109 @@ impl<T: FieldElement> ASMPILConverter<T> {
         Default::default()
     }
 
+    fn compatible(&self, left: &ASMStatement<T>, right: &ASMStatement<T>) -> bool {
```

Comment:
Maybe split some stuff into a new file?

Reply:
Yes I suspect this will not require `self` and can be on an Expression impl